.travis.yml

```diff
@@ -148,6 +148,44 @@ matrix:
       script:
         - py.test tests/webapp/ --runslow
 
+    # Job 6: Python Tests - Selenium integration
+    - env: python-tests-selenium
+      # TODO: Investigate switching back to the container infra, by setting `sudo: false`.
+      sudo: required
+      language: python
+      python: "2.7.13"
+      cache:
+        directories:
+          - ~/venv
+      addons:
+        firefox: latest
+      services:
+        - rabbitmq
+      before_install:
+        - curl -sSo ~/elasticsearch.deb https://download.elastic.co/elasticsearch/release/org/elasticsearch/distribution/deb/elasticsearch/2.3.5/elasticsearch-2.3.5.deb && sudo dpkg -i ~/elasticsearch.deb
+        - sudo service elasticsearch start
+        - sudo cp puppet/files/mysql/my.cnf /etc/mysql/conf.d/treeherder.cnf
+        - sudo service mysql restart
+        # Create a clean virtualenv rather than using the one given to us,
+        # to work around: https://github.com/travis-ci/travis-ci/issues/4873
+        - if [[ ! -f ~/venv/bin/activate ]]; then virtualenv -p python ~/venv; fi
+        - source ~/venv/bin/activate
+        - pip install --disable-pip-version-check --upgrade pip==8.1.1
+        - ./bin/vendor-libmysqlclient.sh ~/venv
+        - wget https://github.com/mozilla/geckodriver/releases/download/v0.14.0/geckodriver-v0.14.0-linux64.tar.gz
+        - mkdir geckodriver
+        - tar -xzf geckodriver-v0.14.0-linux64.tar.gz -C geckodriver
+        - export PATH=$PATH:$PWD/geckodriver
```

Comment:
How about:
```shell
- mkdir $HOME/bin
- curl -sSLf https://github.com/mozilla/geckodriver/releases/download/v0.14.0/geckodriver-v0.14.0-linux64.tar.gz | tar -zxfC $HOME/bin
```
(`$HOME/bin` and `$HOME/.local/bin` are both on `PATH` already; streaming curl into tar means the extraction can happen in parallel)